R/dp_init.R

```diff
@@ -205,11 +224,27 @@ dpconf_init <- function(project_path,
 #' fn_dry(sum(log(1:10)))
 #' }
 #' @export
-fn_dry <- function(fn_called){
+fn_dry <- function(fn_called) {
+
   fn_as_call <- rlang::enexpr(fn_called)
-  dried_fn <- rlang::expr_deparse(fn_as_call, width = 300) # large width avoids line wrapping
+
+  is_creds_set_method <- grepl(pattern = "creds_set_", x = fn_as_call, ignore.case = F)[1]
+
+  if (is_creds_set_method) {
+    fn_as_call_main <- rlang::call_name(fn_as_call)
+    get_fn_args <- rlang::call_args(fn_as_call)
+
+    for (arg in get_fn_args) {
+      if (!rlang::is_callable(arg)) {
+        stop(cli::format_error(glue::glue("Do not supply the credentials directly as the function arguments. Function arguments in {fn_as_call_main} need to be passed as function calls. For example, Sys.getenv()")))
```

Comment:
It might be helpful to be more specific in how to pass credentials via Sys.getenv()

Suggest something more like `key = Sys.getenv(\"API_KEY\")` instead of simply `Sys.getenv()`

Reply:
Done.